app/lib/server/functions/relinquishRoomOwnerships.js

```diff
@@ -0,0 +1,81 @@
+import { hasRole, getUsersInRole, addUserRoles } from '../../../authorization';
+import { FileUpload } from '../../../file-upload';
+import { Users, Subscriptions, Messages, Rooms } from '../../../models';
+
+const bulkRoomCleanUp = (rids) => {
+	// no bulk deletion for files
+	rids.forEach((rid) => FileUpload.removeFilesByRoomId(rid));
+
+	return Promise.await(Promise.all([
+		Subscriptions.removeByRoomIds(rids),
+		Messages.removeByRoomIds(rids),
+		Rooms.removeByIds(rids),
+	]));
+};
+
+export const relinquishRoomOwnerships = function(userId, removeDirectMessages = true) {
```

Comment:
this function should not delete public rooms, it should let public rooms without an owner even if no other user has joined.